Feature: 수강 바구니 관련 기능 구현

ghkdgus29 opened the pull request:
## Key Feature

- 수강 바구니 목록을 조회하는 기능을 구현하였습니다.
- 수강 바구니에 강의를 담는 기능을 구현하였습니다.
  - 수강 바구니에 이미 담긴 과목(Subject)은 다시 담을 수 없습니다.
  - 수강 바구니에 담긴 강의와 시간표가 겹치는 강의는 담을 수 없습니다.
  - 한 학기 기본학점 (18) 을 넘는 강의를 수강바구니에 담을 수 없습니다.
- 수강 바구니에 담긴 강의를 삭제하는 기능을 구현하였습니다.
  - 로그인한 학생이 가지고 있지 않은 강의는 삭제할 수 없습니다.
- 특정 강의를 현재 수강바구니로 담은 학생의 인원수를 조회하는 기능을 구현하였습니다.

## To reviewer

- 수강 바구니의 시간표를 조회하는 기능은 구현하지 않았습니다.
  - 이미 수강 바구니 목록을 반환하는 API 응답에서 시간표를 반환하기 때문입니다.

- 현재 강의 (lecture) 에 학기 기능은 추가되어있지 않습니다.
  - 따라서 Fia 의 PR을 먼저 dev 로 merge 한 후, 저의 PR에서 merge conflict 를 해결하는 step 을 밟아야 할 것 같습니다.
  - merge conflict 를 해결한 후, 수강 바구니에 강의를 담을 때 강의가 현재 학기에 진행되는 강의인지 검증하는 기능을 추가할 예정입니다.

- 현재 Subject 가 department 필드를 갖고 있습니다.
  - 이 역시, merge conflict 를 해결하면서 Lecture 가 department 를 갖도록 변경하면 될 것 같습니다.   

천천히 리뷰 부탁드립니다~! 
감사합니다~!~~!~!

ghkdgus29 committed:
feat: 원하는 수업을 수강 바구니에 담는 기능 구현

ghkdgus29 committed:
feat: 이미 담은 수업은 수강 바구니에 다시 담지 못하도록 검증 기능 구현

ghkdgus29 committed:
modify: 이미 수강 바구니에 담은 과목을 다시 담지 못하도록 검증 기능 구현

- 이전 기능에선 수업 (Lecture) 의 중복여부를 검사하였는데,
- 수업이 달라도 과목 (Subject) 은 같을 수 있으므로
- 과목의 중복여부를 검사하도록 변경

ghkdgus29 committed:
feat: 수강 바구니에 담은 수업들이 한 학기 학점 제한 (18학점) 을 넘지 않도록 검증 기능 구현

ghkdgus29 committed:
refactor: 학생의 수강 바구니 목록들을 DB에서 조회하는 횟수를 2번에서 1번으로 줄임

ghkdgus29 committed:
test: Basket 컨트롤러 테스트 작성

ghkdgus29 committed:
feat: 수강 바구니에 수업을 담을 때, 시간표가 겹치면 담을 수 없도록 검증
기능 구현

ghkdgus29 committed:
feat: 수강 바구니 목록 조회 기능 구현

ghkdgus29 committed:
feat: 수강 바구니 삭제 기능 구현

ghkdgus29 committed:
feat: 특정 강의를 수강 바구니로 담은 학생 수를 조회하는 기능 구현

ghkdgus29 commented:
### Test Coverage Report
|Overall Project|97.09% **`-0.04%`**|:green_apple:|
|:-|:-|:-:|
|Files changed|99.82%|:green_apple:|
<br>

|File|Coverage||
|:-|:-|:-:|
|[BasketService.java](https://github.com/mujik-tigers/course-registration-system/blob/6d1419ec375855dd8531d5638ca11daf1fa1ce83/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Fbasket%2Fapplication%2FBasketService.java)|100%|:green_apple:|
|[ScheduleConflictException.java](https://github.com/mujik-tigers/course-registration-system/blob/6d1419ec375855dd8531d5638ca11daf1fa1ce83/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Fexception%2Fschedule%2FScheduleConflictException.java)|100%|:green_apple:|
|[SelectedLecture.java](https://github.com/mujik-tigers/course-registration-system/blob/6d1419ec375855dd8531d5638ca11daf1fa1ce83/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Fbasket%2Fdto%2FSelectedLecture.java)|100%|:green_apple:|
|[DeletedBasket.java](https://github.com/mujik-tigers/course-registration-system/blob/6d1419ec375855dd8531d5638ca11daf1fa1ce83/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Fbasket%2Fdto%2FDeletedBasket.java)|100%|:green_apple:|
|[BasketDetail.java](https://github.com/mujik-tigers/course-registration-system/blob/6d1419ec375855dd8531d5638ca11daf1fa1ce83/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Fbasket%2Fdto%2FBasketDetail.java)|100%|:green_apple:|
|[BasketList.java](https://github.com/mujik-tigers/course-registration-system/blob/6d1419ec375855dd8531d5638ca11daf1fa1ce83/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Fbasket%2Fdto%2FBasketList.java)|100%|:green_apple:|
|[StudentService.java](https://github.com/mujik-tigers/course-registration-system/blob/6d1419ec375855dd8531d5638ca11daf1fa1ce83/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Fstudent%2Fapplication%2FStudentService.java)|100%|:green_apple:|
|[BasketStoringCount.java](https://github.com/mujik-tigers/course-registration-system/blob/6d1419ec375855dd8531d5638ca11daf1fa1ce83/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Flecture%2Fdto%2FBasketStoringCount.java)|100%|:green_apple:|
|[LectureController.java](https://github.com/mujik-tigers/course-registration-system/blob/6d1419ec375855dd8531d5638ca11daf1fa1ce83/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Flecture%2Fpresentation%2FLectureController.java)|100%|:green_apple:|
|[ResponseMessage.java](https://github.com/mujik-tigers/course-registration-system/blob/6d1419ec375855dd8531d5638ca11daf1fa1ce83/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Futil%2Fapi%2FResponseMessage.java)|100%|:green_apple:|
|[LectureService.java](https://github.com/mujik-tigers/course-registration-system/blob/6d1419ec375855dd8531d5638ca11daf1fa1ce83/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Flecture%2Fapplication%2FLectureService.java)|100%|:green_apple:|
|[AuthService.java](https://github.com/mujik-tigers/course-registration-system/blob/6d1419ec375855dd8531d5638ca11daf1fa1ce83/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Fauth%2Fapplication%2FAuthService.java)|100%|:green_apple:|
|[Basket.java](https://github.com/mujik-tigers/course-registration-system/blob/6d1419ec375855dd8531d5638ca11daf1fa1ce83/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Fbasket%2FBasket.java)|100%|:green_apple:|
|[BasketController.java](https://github.com/mujik-tigers/course-registration-system/blob/6d1419ec375855dd8531d5638ca11daf1fa1ce83/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Fbasket%2Fpresentation%2FBasketController.java)|100%|:green_apple:|
|[ErrorType.java](https://github.com/mujik-tigers/course-registration-system/blob/6d1419ec375855dd8531d5638ca11daf1fa1ce83/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Fexception%2FErrorType.java)|100%|:green_apple:|
|[DayOfWeek.java](https://github.com/mujik-tigers/course-registration-system/blob/6d1419ec375855dd8531d5638ca11daf1fa1ce83/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Fschedule%2FDayOfWeek.java)|100%|:green_apple:|
|[NonexistenceLectureException.java](https://github.com/mujik-tigers/course-registration-system/blob/6d1419ec375855dd8531d5638ca11daf1fa1ce83/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Fexception%2Flecture%2FNonexistenceLectureException.java)|100%|:green_apple:|
|[DuplicateBasketException.java](https://github.com/mujik-tigers/course-registration-system/blob/6d1419ec375855dd8531d5638ca11daf1fa1ce83/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Fexception%2Fbasket%2FDuplicateBasketException.java)|100%|:green_apple:|
|[ExceededCreditLimitException.java](https://github.com/mujik-tigers/course-registration-system/blob/6d1419ec375855dd8531d5638ca11daf1fa1ce83/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Fexception%2Fbasket%2FExceededCreditLimitException.java)|100%|:green_apple:|
|[NonexistenceBasketException.java](https://github.com/mujik-tigers/course-registration-system/blob/6d1419ec375855dd8531d5638ca11daf1fa1ce83/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Fexception%2Fbasket%2FNonexistenceBasketException.java)|100%|:green_apple:|
|[Enrollment.java](https://github.com/mujik-tigers/course-registration-system/blob/6d1419ec375855dd8531d5638ca11daf1fa1ce83/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Fenrollment%2FEnrollment.java)|100%|:green_apple:|
|[Lecture.java](https://github.com/mujik-tigers/course-registration-system/blob/6d1419ec375855dd8531d5638ca11daf1fa1ce83/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Flecture%2FLecture.java)|95.56%|:green_apple:|
|[Schedule.java](https://github.com/mujik-tigers/course-registration-system/blob/6d1419ec375855dd8531d5638ca11daf1fa1ce83/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Fschedule%2FSchedule.java)|90.79% **`-1.32%`**|:green_apple:|
|[Student.java](https://github.com/mujik-tigers/course-registration-system/blob/6d1419ec375855dd8531d5638ca11daf1fa1ce83/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Fstudent%2FStudent.java)|88.24%|:green_apple:|
|[ProjectConstant.java](https://github.com/mujik-tigers/course-registration-system/blob/6d1419ec375855dd8531d5638ca11daf1fa1ce83/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Futil%2FProjectConstant.java)|0%|:green_apple:|

yeonise commented on src/main/java/site/courseregistrationsystem/basket/infrastructure/BasketRepository.java:
```diff
@@ -0,0 +1,27 @@
+
+	Optional<Basket> findByIdAndStudent(Long basketId, Student student);
+
+	@Modifying
+	@Query("delete from Basket b where b = :basket")
+	void delete(@Param("basket") Basket basket);
```
💁🏻 `delete()` 를 직접 생성하며 `@Query` 를 적어준 이유가 궁금해요

ghkdgus29 replied:
deleteByXX 메서드를 사용하면은 조회쿼리가 나가게 되는데 어차피 이전에 검증을 하니까 순수한 delete 쿼리만 나가도록 하고 싶었어요!

yeonise replied:
👍🏻 더 최적화할 수 있군요!

yeonise commented on src/main/java/site/courseregistrationsystem/basket/presentation/BasketController.java:
```diff
@@ -0,0 +1,47 @@
+		return ApiResponse.ok(ResponseMessage.BASKET_ADD_SUCCESS.getMessage(), selectedLecture);
+	}
+
+	@GetMapping("/baskets")
+	public ApiResponse<BasketList> fetchBaskets(@Login Long studentPk) {
+		BasketList basketList = basketService.fetchBaskets(studentPk);
+
+		return ApiResponse.ok(ResponseMessage.BASKET_FETCH_SUCCESS.getMessage(), basketList);
+	}
+
+	@DeleteMapping("/baskets/{basketId}")
+	public ApiResponse<DeletedBasket> deleteBasket(@Login Long studentPk, @PathVariable Long basketId) {
+		Long deletedBasketId = basketService.deleteBasket(studentPk, basketId);
+		DeletedBasket deletedBasket = new DeletedBasket(deletedBasketId);
+
+		return ApiResponse.ok(ResponseMessage.BASKET_DELETE_SUCCESS.getMessage(), deletedBasket);
+	}
```
💁🏻 일부 DTO는 service에서 생성하여 반환하고, 일부 DTO는 controller에서 생성하는 데 생성 위치가 다른 이유가 궁금해요

ghkdgus29 replied:
왠만하면 컨트롤러에서 만들어서 반환하려고 하는데, 그게 힘든 경우에 (반환하는 값이 2개 이상인 경우) 는 서비스에서 만들어서 반환하고 있어요!

yeonise commented on src/main/java/site/courseregistrationsystem/basket/infrastructure/BasketRepository.java:
```diff
@@ -0,0 +1,27 @@
+
+	List<Basket> findAllByStudent(Student student);
+
+	Optional<Basket> findByIdAndStudent(Long basketId, Student student);
```
💁🏻 `findById` 와 `findByIdAndStudent` 가 동일한 결과를 가져옴에도 `AndStudent` 를 사용한 건 자신의 Basket인지 검증까지 한번에 하기 위해서라고 이해했습니다.
``` java
Basket basket = basketRepository.findByIdAndStudent(basketId, student)
			.orElseThrow(NonexistenceBasketException::new);
```
저는 가져온 뒤 Student가 일치하지 않으면 **401 Unauthorization**을 응답하도록 구현했는데, 나중에 같이 얘기해보면 좋을 것 같아요!

ghkdgus29 replied:
좀 더 디테일하게 응답메시지를 주는 거니까 좋은 거 같아요!

yeonise commented on src/main/java/site/courseregistrationsystem/basket/infrastructure/BasketRepository.java:
```diff
@@ -0,0 +1,27 @@
+
+public interface BasketRepository extends JpaRepository<Basket, Long> {
+
+	List<Basket> findAllByStudent(Student student);
```
💁🏻 `fetch student` 혹은 `fetch lecture` 를 적어주지 않아도 이후에 쿼리가 추가로 발생하지 않는지 궁금해요

ghkdgus29 replied:
그 부분을 생각 못했네요! 
해당 부분은 fetch join 하는게 나을거같아요!

yeonise commented on src/main/java/site/courseregistrationsystem/lecture/presentation/LectureController.java:
```diff
@@ -25,4 +27,11 @@ public ApiResponse<LectureSchedulePage> fetchLectureSchedule(
 			lectureService.fetchLectureSchedule(pageable, lectureFilterOptions));
 	}
 
+	@GetMapping("/lectures/{lectureId}/basket-count")
+	public ApiResponse<BasketStoringCount> fetchBasketStoringCount(@PathVariable Long lectureId) {
+		BasketStoringCount basketStoringCount = lectureService.fetchBasketStoringCount(lectureId);
+
+		return ApiResponse.ok(ResponseMessage.BASKET_STORING_COUNT_FETCH_SUCCESS.getMessage(), basketStoringCount);
+	}
+
```
💁🏻 도메인을 Basket에서 Lecture로 변경한 이유가 궁금해요

ghkdgus29 replied:
Basket 보다는 Lecture 에서 현재 수강바구니에 담은 인원수를 조회하는 기능을 제공하는 것이 자연스러운 것 같아서요!

yeonise commented on src/main/java/site/courseregistrationsystem/lecture/application/LectureService.java:
```diff
@@ -27,4 +35,13 @@ public LectureSchedulePage fetchLectureSchedule(Pageable pageable, LectureFilter
+		Lecture lecture = lectureRepository.findById(lectureId)
+			.orElseThrow(NonexistenceLectureException::new);
+
+		List<Basket> baskets = basketRepository.findAllByLecture(lecture);
```
💁🏻 `count` 쿼리 대신 `findAll` 을 사용한 이유가 궁금해요

ghkdgus29 replied:
특별한 이유는 없어요 ㅋㅋ 
count 쿼리로 하는것이 낫겠네요

yeonise reviewed (approved):
이번 주도 수고하셨습니다! 굿굿~! 🚀

yeonise commented on src/test/java/site/courseregistrationsystem/basket/application/BasketServiceTest.java:
```diff
@@ -0,0 +1,330 @@
+	private EntityManager entityManager;
+
+	@Test
+	@DisplayName("학생은 원하는 수업을 선택하여 수강 바구니에 담은 뒤, 담은 수업(Lecture)의 id 를 반환한다.")
```
문장에서 학생이 id를 반환하는 것처럼 보이기도 하네욤!
💁🏻 요청한 학생의 수강 바구니에 수업을 성공적으로 담으면 해당 수업의 ID를 반환한다~..

ghkdgus29 replied:
밑에 문장이 더 자연스러운 것 같네요~! 
수정할게용

yeonise commented on src/test/java/site/courseregistrationsystem/basket/application/BasketServiceTest.java:
```diff
@@ -0,0 +1,330 @@
+		int LECTURE_COUNT = 6;
+
+		// when
+		for (int i = 0; i < LECTURE_COUNT; i++) {                                        // 주어진 학생이 18학점의 수업을 수강바구니에 담음
```
주석 띄어쓰기가 이상한 것 같아요!

ghkdgus29 replied:
ㅋㅋㅋㅋㅋㅋㅋㅋㅋ

yeonise commented on src/test/java/site/courseregistrationsystem/basket/application/BasketServiceTest.java:
```diff
@@ -0,0 +1,330 @@
+	}
+
+	@Test
+	@DisplayName("수강 바구니에 담은 수업둘 목록을 가져올 수 있다.")
+	void fetchBasketList() throws Exception {
```
오타가 있네요! 둘 → 들

ghkdgus29 replied:
ㅋㅋㅋㅋ 수정할게요~!

ghkdgus29 committed:
fix: PR 리뷰에서의 요구사항들을 반영

- 페치 조인 사용하여 효율성 높이기
- count 쿼리를 날려 효율성 높이기
- 테스트 코드의 @DisplayName 을 수정하여 가독성 높이기

ghkdgus29 commented:
### Test Coverage Report
|Overall Project|97.09% **`-0.04%`**|:green_apple:|
|:-|:-|:-:|
|Files changed|99.82%|:green_apple:|
<br>

|File|Coverage||
|:-|:-|:-:|
|[BasketService.java](https://github.com/mujik-tigers/course-registration-system/blob/d1a725bd61843b5c5a850daea580185e5ede528e/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Fbasket%2Fapplication%2FBasketService.java)|100%|:green_apple:|
|[ScheduleConflictException.java](https://github.com/mujik-tigers/course-registration-system/blob/d1a725bd61843b5c5a850daea580185e5ede528e/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Fexception%2Fschedule%2FScheduleConflictException.java)|100%|:green_apple:|
|[SelectedLecture.java](https://github.com/mujik-tigers/course-registration-system/blob/d1a725bd61843b5c5a850daea580185e5ede528e/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Fbasket%2Fdto%2FSelectedLecture.java)|100%|:green_apple:|
|[DeletedBasket.java](https://github.com/mujik-tigers/course-registration-system/blob/d1a725bd61843b5c5a850daea580185e5ede528e/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Fbasket%2Fdto%2FDeletedBasket.java)|100%|:green_apple:|
|[BasketDetail.java](https://github.com/mujik-tigers/course-registration-system/blob/d1a725bd61843b5c5a850daea580185e5ede528e/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Fbasket%2Fdto%2FBasketDetail.java)|100%|:green_apple:|
|[BasketList.java](https://github.com/mujik-tigers/course-registration-system/blob/d1a725bd61843b5c5a850daea580185e5ede528e/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Fbasket%2Fdto%2FBasketList.java)|100%|:green_apple:|
|[StudentService.java](https://github.com/mujik-tigers/course-registration-system/blob/d1a725bd61843b5c5a850daea580185e5ede528e/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Fstudent%2Fapplication%2FStudentService.java)|100%|:green_apple:|
|[BasketStoringCount.java](https://github.com/mujik-tigers/course-registration-system/blob/d1a725bd61843b5c5a850daea580185e5ede528e/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Flecture%2Fdto%2FBasketStoringCount.java)|100%|:green_apple:|
|[LectureController.java](https://github.com/mujik-tigers/course-registration-system/blob/d1a725bd61843b5c5a850daea580185e5ede528e/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Flecture%2Fpresentation%2FLectureController.java)|100%|:green_apple:|
|[ResponseMessage.java](https://github.com/mujik-tigers/course-registration-system/blob/d1a725bd61843b5c5a850daea580185e5ede528e/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Futil%2Fapi%2FResponseMessage.java)|100%|:green_apple:|
|[LectureService.java](https://github.com/mujik-tigers/course-registration-system/blob/d1a725bd61843b5c5a850daea580185e5ede528e/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Flecture%2Fapplication%2FLectureService.java)|100%|:green_apple:|
|[AuthService.java](https://github.com/mujik-tigers/course-registration-system/blob/d1a725bd61843b5c5a850daea580185e5ede528e/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Fauth%2Fapplication%2FAuthService.java)|100%|:green_apple:|
|[Basket.java](https://github.com/mujik-tigers/course-registration-system/blob/d1a725bd61843b5c5a850daea580185e5ede528e/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Fbasket%2FBasket.java)|100%|:green_apple:|
|[BasketController.java](https://github.com/mujik-tigers/course-registration-system/blob/d1a725bd61843b5c5a850daea580185e5ede528e/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Fbasket%2Fpresentation%2FBasketController.java)|100%|:green_apple:|
|[ErrorType.java](https://github.com/mujik-tigers/course-registration-system/blob/d1a725bd61843b5c5a850daea580185e5ede528e/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Fexception%2FErrorType.java)|100%|:green_apple:|
|[DayOfWeek.java](https://github.com/mujik-tigers/course-registration-system/blob/d1a725bd61843b5c5a850daea580185e5ede528e/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Fschedule%2FDayOfWeek.java)|100%|:green_apple:|
|[NonexistenceLectureException.java](https://github.com/mujik-tigers/course-registration-system/blob/d1a725bd61843b5c5a850daea580185e5ede528e/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Fexception%2Flecture%2FNonexistenceLectureException.java)|100%|:green_apple:|
|[DuplicateBasketException.java](https://github.com/mujik-tigers/course-registration-system/blob/d1a725bd61843b5c5a850daea580185e5ede528e/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Fexception%2Fbasket%2FDuplicateBasketException.java)|100%|:green_apple:|
|[ExceededCreditLimitException.java](https://github.com/mujik-tigers/course-registration-system/blob/d1a725bd61843b5c5a850daea580185e5ede528e/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Fexception%2Fbasket%2FExceededCreditLimitException.java)|100%|:green_apple:|
|[NonexistenceBasketException.java](https://github.com/mujik-tigers/course-registration-system/blob/d1a725bd61843b5c5a850daea580185e5ede528e/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Fexception%2Fbasket%2FNonexistenceBasketException.java)|100%|:green_apple:|
|[Enrollment.java](https://github.com/mujik-tigers/course-registration-system/blob/d1a725bd61843b5c5a850daea580185e5ede528e/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Fenrollment%2FEnrollment.java)|100%|:green_apple:|
|[Lecture.java](https://github.com/mujik-tigers/course-registration-system/blob/d1a725bd61843b5c5a850daea580185e5ede528e/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Flecture%2FLecture.java)|95.56%|:green_apple:|
|[Schedule.java](https://github.com/mujik-tigers/course-registration-system/blob/d1a725bd61843b5c5a850daea580185e5ede528e/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Fschedule%2FSchedule.java)|90.79% **`-1.32%`**|:green_apple:|
|[Student.java](https://github.com/mujik-tigers/course-registration-system/blob/d1a725bd61843b5c5a850daea580185e5ede528e/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Fstudent%2FStudent.java)|88.24%|:green_apple:|
|[ProjectConstant.java](https://github.com/mujik-tigers/course-registration-system/blob/d1a725bd61843b5c5a850daea580185e5ede528e/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Futil%2FProjectConstant.java)|0%|:green_apple:|

ghkdgus29 committed:
Merge branch 'dev' into feature/20-basket-page

ghkdgus29 committed:
feat: 수강 바구니에 수업을 담거나, 특정 강의를 수강 바구니에 담은 학생 수 조회 시, 해당 수업이 현재 신청 학기에 해당하는지 검증 기능 구현

ghkdgus29 commented:
### Test Coverage Report
|Overall Project|96.38% **`-0.87%`**|:green_apple:|
|:-|:-|:-:|
|Files changed|95.61%|:green_apple:|
<br>

|File|Coverage||
|:-|:-|:-:|
|[BasketService.java](https://github.com/mujik-tigers/course-registration-system/blob/12237330c37bc61d106370eccdd515db9c525c42/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Fbasket%2Fapplication%2FBasketService.java)|100%|:green_apple:|
|[ScheduleConflictException.java](https://github.com/mujik-tigers/course-registration-system/blob/12237330c37bc61d106370eccdd515db9c525c42/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Fexception%2Fschedule%2FScheduleConflictException.java)|100%|:green_apple:|
|[SelectedLecture.java](https://github.com/mujik-tigers/course-registration-system/blob/12237330c37bc61d106370eccdd515db9c525c42/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Fbasket%2Fdto%2FSelectedLecture.java)|100%|:green_apple:|
|[DeletedBasket.java](https://github.com/mujik-tigers/course-registration-system/blob/12237330c37bc61d106370eccdd515db9c525c42/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Fbasket%2Fdto%2FDeletedBasket.java)|100%|:green_apple:|
|[BasketDetail.java](https://github.com/mujik-tigers/course-registration-system/blob/12237330c37bc61d106370eccdd515db9c525c42/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Fbasket%2Fdto%2FBasketDetail.java)|100%|:green_apple:|
|[BasketList.java](https://github.com/mujik-tigers/course-registration-system/blob/12237330c37bc61d106370eccdd515db9c525c42/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Fbasket%2Fdto%2FBasketList.java)|100%|:green_apple:|
|[SemesterInvalidException.java](https://github.com/mujik-tigers/course-registration-system/blob/12237330c37bc61d106370eccdd515db9c525c42/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Fexception%2Fsemester%2FSemesterInvalidException.java)|100%|:green_apple:|
|[StudentService.java](https://github.com/mujik-tigers/course-registration-system/blob/12237330c37bc61d106370eccdd515db9c525c42/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Fstudent%2Fapplication%2FStudentService.java)|100%|:green_apple:|
|[BasketStoringCount.java](https://github.com/mujik-tigers/course-registration-system/blob/12237330c37bc61d106370eccdd515db9c525c42/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Flecture%2Fdto%2FBasketStoringCount.java)|100%|:green_apple:|
|[LectureController.java](https://github.com/mujik-tigers/course-registration-system/blob/12237330c37bc61d106370eccdd515db9c525c42/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Flecture%2Fpresentation%2FLectureController.java)|100%|:green_apple:|
|[ResponseMessage.java](https://github.com/mujik-tigers/course-registration-system/blob/12237330c37bc61d106370eccdd515db9c525c42/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Futil%2Fapi%2FResponseMessage.java)|100%|:green_apple:|
|[Basket.java](https://github.com/mujik-tigers/course-registration-system/blob/12237330c37bc61d106370eccdd515db9c525c42/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Fbasket%2FBasket.java)|100%|:green_apple:|
|[BasketController.java](https://github.com/mujik-tigers/course-registration-system/blob/12237330c37bc61d106370eccdd515db9c525c42/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Fbasket%2Fpresentation%2FBasketController.java)|100%|:green_apple:|
|[CreditLimitExceededException.java](https://github.com/mujik-tigers/course-registration-system/blob/12237330c37bc61d106370eccdd515db9c525c42/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Fexception%2Fcredit%2FCreditLimitExceededException.java)|100%|:green_apple:|
|[ErrorType.java](https://github.com/mujik-tigers/course-registration-system/blob/12237330c37bc61d106370eccdd515db9c525c42/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Fexception%2FErrorType.java)|100%|:green_apple:|
|[DayOfWeek.java](https://github.com/mujik-tigers/course-registration-system/blob/12237330c37bc61d106370eccdd515db9c525c42/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Fschedule%2FDayOfWeek.java)|100%|:green_apple:|
|[DuplicateBasketException.java](https://github.com/mujik-tigers/course-registration-system/blob/12237330c37bc61d106370eccdd515db9c525c42/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Fexception%2Fbasket%2FDuplicateBasketException.java)|100%|:green_apple:|
|[NonexistenceBasketException.java](https://github.com/mujik-tigers/course-registration-system/blob/12237330c37bc61d106370eccdd515db9c525c42/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Fexception%2Fbasket%2FNonexistenceBasketException.java)|100%|:green_apple:|
|[EnrollmentService.java](https://github.com/mujik-tigers/course-registration-system/blob/12237330c37bc61d106370eccdd515db9c525c42/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Fenrollment%2Fapplication%2FEnrollmentService.java)|100%|:green_apple:|
|[LectureService.java](https://github.com/mujik-tigers/course-registration-system/blob/12237330c37bc61d106370eccdd515db9c525c42/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Flecture%2Fapplication%2FLectureService.java)|93.85% **`-6.15%`**|:green_apple:|
|[Schedule.java](https://github.com/mujik-tigers/course-registration-system/blob/12237330c37bc61d106370eccdd515db9c525c42/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Fschedule%2FSchedule.java)|92.31%|:green_apple:|
|[Enrollment.java](https://github.com/mujik-tigers/course-registration-system/blob/12237330c37bc61d106370eccdd515db9c525c42/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Fenrollment%2FEnrollment.java)|91.18%|:green_apple:|
|[Lecture.java](https://github.com/mujik-tigers/course-registration-system/blob/12237330c37bc61d106370eccdd515db9c525c42/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Flecture%2FLecture.java)|88.82%|:green_apple:|
|[Student.java](https://github.com/mujik-tigers/course-registration-system/blob/12237330c37bc61d106370eccdd515db9c525c42/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Fstudent%2FStudent.java)|88.24%|:green_apple:|
|[Semester.java](https://github.com/mujik-tigers/course-registration-system/blob/12237330c37bc61d106370eccdd515db9c525c42/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Flecture%2FSemester.java)|39.47% **`-60.53%`**|:x:|
|[ProjectConstant.java](https://github.com/mujik-tigers/course-registration-system/blob/12237330c37bc61d106370eccdd515db9c525c42/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Futil%2FProjectConstant.java)|0%|:green_apple:|
